### ISSUE TITLE
Issue with V100 Distributed Training

### ISSUE THREAD

**Issue by pablogranolabar** [guxd/DialogBERT#7]
I have the following distributed training setup working without issue on Tesla K80, but whenever I attempt to do this with an 8X V100 the training process just silently hangs without dispatching any process to any of the GPUs:

```
export MASTER_PORT=29500
export MASTER_ADDR="127.0.0.1"
export WORLD_SIZE=8
export RANK=0
python3 main.py --model_size=large --per_gpu_train_batch_size=128 --local_rank 0
```
What's weird is that training works fine on a single GPU if I drop the --local_rank flag. While the process is just hanging, nothing is being dispatched to any of the GPUs:

```
$ sudo nvidia-smi
Sat May 15 21:53:13 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.119.04   Driver Version: 450.119.04   CUDA Version: 11.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  A100-SXM4-40GB      On   | 00000000:10:1C.0 Off |                    0 |
| N/A   52C    P0    60W / 400W |      3MiB / 40537MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   1  A100-SXM4-40GB      On   | 00000000:10:1D.0 Off |                    0 |
| N/A   47C    P0    56W / 400W |      3MiB / 40537MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   2  A100-SXM4-40GB      On   | 00000000:20:1C.0 Off |                    0 |
| N/A   49C    P0    56W / 400W |      3MiB / 40537MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   3  A100-SXM4-40GB      On   | 00000000:20:1D.0 Off |                    0 |
| N/A   45C    P0    55W / 400W |      3MiB / 40537MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   4  A100-SXM4-40GB      On   | 00000000:90:1C.0 Off |                    0 |
| N/A   50C    P0    56W / 400W |      3MiB / 40537MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   5  A100-SXM4-40GB      On   | 00000000:90:1D.0 Off |                    0 |
| N/A   45C    P0    56W / 400W |      3MiB / 40537MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   6  A100-SXM4-40GB      On   | 00000000:A0:1C.0 Off |                    0 |
| N/A   52C    P0    60W / 400W |      3MiB / 40537MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   7  A100-SXM4-40GB      On   | 00000000:A0:1D.0 Off |                    0 |
| N/A   48C    P0    57W / 400W |      3MiB / 40537MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```
Any ideas?

**Comment by guxd** [comment]
Sorry I have no idea about this issue. You may refer to the latest multi-gpu examples in Hugginface Transformer (https://github.com/huggingface/transformers) or PyTorch documentation.
Or you can try debugging tools to identify the location that causes the silent hang.

**Comment by pablogranolabar** [comment]
So I've had this training on a V100 for about 6 days now using:

```
python3 main.py --model_size=large --per_gpu_train_batch_size=128
```
But the loss numbers seem very high, a couple of stats from the last few epochs:

```
avg_len = 8.290504451038576
bleu = 0.008627823349932331
meteor = 0.014166421098098157
nist = 0.023135396644037475
perplexity = 561.353515625
rouge-L = 0.025940259012983104
valid_loss = 6.330350998813391


avg_len = 7.5495826849625125
bleu = 0.006454373723218462
meteor = 0.011782214090490118
nist = 0.015940440749686976
perplexity = 490.2471008300781
rouge-L = 0.02181344721801129
valid_loss = 6.194909538233157
```
Does this indicate that the model is failing to converge or am I looking at several months of training with this? Do you mind sharing your final numbers for your trained model for comparison?

**Comment by guxd** [comment]
This seems like the phenomenon of overfitting..
For comparisons with ours, you should specify your model size as `tiny`.
